static/js/typeahead_helper.js

```diff
@@ -207,17 +207,37 @@ exports.compare_by_sub_count = function (stream_a, stream_b) {
     return stream_a.subscribers.num_items() < stream_b.subscribers.num_items();
 };
 
+function activity_score(stream) {
+    var stream_score = 0;
+    if (stream.pin_to_top) {
+        stream_score += 2;
```

Comment:
We can simplify this function to just explicitly return 2, 1, and 0 for the scores.  No need for `+=`.